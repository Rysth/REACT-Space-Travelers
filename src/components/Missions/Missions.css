:root {
  --CL-primary-gray: #c7c7c7;
  --CL-primary-gray-darker: #868686;
  --CL-primary-light: #e0e0e0;
}

.missions-content {
  padding: 1em 0;
  overflow: auto;
}

.missions-table {
  width: max-content;
}

.missions-title {
  font-size: 0.95rem;
  font-weight: bold;
}

.missions-title,
.missions-data {
  border: 1px solid var(--CL-primary-gray);
  padding: 1em;
}

.missions-heading {
  display: grid;
  grid-template-columns: 150px 600px 180px 180px;
}

@media (min-width: 768px) {
  .missions-heading {
    display: grid;
    grid-template-columns: 150px 890px 180px 180px;
  }
}
