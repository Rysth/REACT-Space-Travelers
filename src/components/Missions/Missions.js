import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchMissions } from '../../redux/missions/missionsSlice';
import MissionItem from './Item/MissionItem';
import './Missions.css';

function Missions() {
  const { missions } = useSelector((store) => store.missions);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchMissions());
  }, [dispatch]);

  return (
    <div className="missions">
      <div className="missions-content container">
        <div className="missions-table">
          <header className="missions-heading">
            <h2 className="missions-title">Missions</h2>
            <h2 className="missions-title">Description</h2>
            <h2 className="missions-title">Status</h2>
            <span className="missions-title" />
          </header>
          <div className="missions-body">
            {missions.map((mission, index) => (
              <MissionItem
                key={mission.id}
                id={mission.id}
                index={index}
                title={mission.name}
                parragraph={mission.description}
                reserved={mission.reserved}
              />
            ))}
          </div>
        </div>
      </div>
    </div>
  );
}

export default Missions;
