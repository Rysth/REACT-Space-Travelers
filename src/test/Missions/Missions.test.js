import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import store from '../../redux/Store';
import Missions from '../../components/Missions/Missions';

describe('Add Test', () => {
  const missionsContainer = (
    <Provider store={store}>
      <Missions />
    </Provider>
  );

  test('Take snapshot of this container', () => {
    const container = render(missionsContainer);
    expect(container).toMatchSnapshot();
  });
});
