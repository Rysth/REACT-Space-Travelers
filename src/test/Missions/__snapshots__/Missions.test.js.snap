// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Test Take snapshot of this container 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="missions"
      >
        <div
          class="missions-content container"
        >
          <div
            class="missions-table"
          >
            <header
              class="missions-heading"
            >
              <h2
                class="missions-title"
              >
                Missions
              </h2>
              <h2
                class="missions-title"
              >
                Description
              </h2>
              <h2
                class="missions-title"
              >
                Status
              </h2>
              <span
                class="missions-title"
              />
            </header>
            <div
              class="missions-body"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="missions"
    >
      <div
        class="missions-content container"
      >
        <div
          class="missions-table"
        >
          <header
            class="missions-heading"
          >
            <h2
              class="missions-title"
            >
              Missions
            </h2>
            <h2
              class="missions-title"
            >
              Description
            </h2>
            <h2
              class="missions-title"
            >
              Status
            </h2>
            <span
              class="missions-title"
            />
          </header>
          <div
            class="missions-body"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
